src/Connection/Cluster/PredisCluster.php

```diff
@@ -285,4 +297,36 @@ public function executeCommandOnEachNode(CommandInterface $command): array
 
         return $responses;
     }
+
+    /**
+     * Loop over connections until there's data to read.
+     *
+     * @return mixed
+     */
+    public function read()
```

Comment:
Maybe it's not a use case here, but I would feel better if I could provide a maximum timeout for this read, so it does not get stuck in an endless loop in case who knows what happens to the connections.